vars/waitForHelixRuns.groovy

```diff
@@ -24,24 +24,26 @@ def call (def helixRunsBlob, String prStatusPrefix) {
     def failedRunMap = [:]
     def passed = true
 
-    for (int i = 0; i < helixRunsBlob.size(); i++) {
-        try {
-            def currentRun = helixRunsBlob[i];
-            def queueId = currentRun['QueueId']
-            def correlationId = currentRun['CorrelationId']
-            def statusUrl = "https://helix.dot.net/api/2017-04-14/jobs/${correlationId}/details"
-            def statusResponse = httpRequest consoleLogResponseBody: true, url: statusUrl
-            assert statusResponse != null
-            assert statusResponse.content != null
-            def statusContent = (new JsonSlurperClassic()).parseText(statusResponse.content)
-            def mcResultsUrl = "https://mc.dot.net/#/user/${getEncodedUrl(statusContent.Creator)}/${getEncodedUrl(statusContent.Source)}/${getEncodedUrl(statusContent.Type)}/${getEncodedUrl(statusContent.Build)}"
-            mcUrlMap[queueId] = mcResultsUrl
-        }
-        catch(Exception ex) {
-            passed = false
-            println(ex.toString());
-            println(ex.getMessage());
-            println(ex.getStackTrace());
+    // Wait until the Helix runs register as started. Wait up to 3 mins between checks
+    waitUntil (minRecurrencePeriod: 120, maxRecurrencePeriod: 180, unit: 'SECONDS') {
+        for (int i = 0; i < helixRunsBlob.size(); i++) {
+            try {
+                def currentRun = helixRunsBlob[i];
+                def queueId = currentRun['QueueId']
+                def correlationId = currentRun['CorrelationId']
+                def statusUrl = "https://helix.dot.net/api/2017-04-14/jobs/${correlationId}/details"
+                def statusResponse = httpRequest consoleLogResponseBody: true, url: statusUrl
+                assert statusResponse != null
+                assert statusResponse.content != null
+                def statusContent = (new JsonSlurperClassic()).parseText(statusResponse.content)
+                def mcResultsUrl = "https://mc.dot.net/#/user/${getEncodedUrl(statusContent.Creator)}/${getEncodedUrl(statusContent.Source)}/${getEncodedUrl(statusContent.Type)}/${getEncodedUrl(statusContent.Build)}"
+                mcUrlMap[queueId] = mcResultsUrl
+            }
+            catch(Exception ex) {
+                println(ex.toString());
```

Comment:
waitUntil expects that the block will return false until it's done, at which time it returns true. So you want to do one of the following:

1) Wrap the inner part of the for in the waitUntil, and return false from the catch and true from the end of the true
2) Wrap the for in the waitUnitl and track whether you have the mcUrlMap filled, returning false until you do.